### PR TITLE
EOS-18387: UDX registration and data movement should be working on OVA

### PR DESCRIPTION
**Pull request by tb-s**
# Backend

## Problem Statement

[EOS-18387: UDX registration and data movement should be working on OVA](https://jts.seagate.com/browse/EOS-18387)
[EOS-18415: Update `csm_setup` with required changes for Lyve Pilot registration and data move on OVA](https://jts.seagate.com/browse/EOS-18415)
[EOS-18416: Update `csm_agent` with required changes for Lyve Pilot registration and data move on OVA](https://jts.seagate.com/browse/EOS-18416)

See tickets for details.

Also, please notice this is not a fully working solution.  Please refer to [EOS-18409](https://jts.seagate.com/browse/EOS-18409) for more details.

## Unit testing on RPM done

Yes.

## Problem Description

See tickets for details.

## Solution

- ~Modify `csm_setup` to support UDS public IP on VM setups~ (1st attempt, original PR)
  * ~Fix minor quirks during UDS configuration step~ (1st attempt, original PR)
- ~Fix `UDS.public_ip` configuration key in `UDSConfigGenerator` with new configuration key format (`UDS>public_ip`)~ (1st attempt, original PR)
- ~Use UDS public IP both as data IP and management IP within USL (in R1/HW setups it was being used only as the data IP)~ (1st attempt, original PR)
- Remove UDS configuration logic from `csm_setup`
- Create network addresses helper
  * Read virtual host IP from `PROVISIONER>virtual_host` (R2 provisioning spec key: `cluster>{cluster-id}>network>management>virtual_host`)
  * Read node's public data domain name from `PROVISIONER>node_public_data_domain_name` (R2 provisioning spec key: `server_node>{machine-id}>network>data>public_fqdn` and obtain the IP address via DNS lookup
- Replace uses of provisioner to obtain network address info by new network addresses help
  * On USL service
  * On S3 server info service
- Fix S3 server information service initialization bug in `csm/core/controllers/s3/server_info.py`

## Test Cases

Should be tested only after required fixes described in [EOS-18409](https://jts.seagate.com/browse/EOS-18409) are in place.  Unit test cases are:
- Lyve Pilot registration succeeds
- Lyve Pilot data move succeeds

### CODE REVIEW

**Comment by tb-s** [comment]
VM Deployment checks are failing due to an `sspl_config` error.

File: csm/conf/configure.py [hunk 1]
```diff
@@ -53,11 +53,11 @@ async def execute(self, command):
             Log.error(f"Configuration Loading Failed {e}")
         self._set_deployment_mode()
         try:
+            uds_public_ip = command.options.get('uds_public_ip')
```
Comment (kupranay) [inline csm/conf/configure.py:56]: Mostly this section should not be required now. it should be taken from configuration file. Lets discuss it in detail to understand both side (provisioning restriction and UDS configuration) better.

**Review by kupranay — CHANGES_REQUESTED** [review]
Marking it as request changes to discuss more

File: csm/conf/configure.py (hunk 1 above)
Comment (tb-s) [reply csm/conf/configure.py]: This PR has been updated, please review again.

**Comment by kupranay** [comment]
ok to test

**Review by kupranay — APPROVED** [review]

**Review by avrnv — APPROVED** [review]

**Pull request merged**